shap/_explanation.py

```diff
@@ -665,8 +656,8 @@ def hstack(self, other: Explanation) -> Explanation:
 
         """
         assert self.shape[0] == other.shape[0], "Can't hstack explanations with different numbers of rows!"
-        assert (
-            np.max(np.abs(self.base_values - other.base_values)) < 1e-6
+        assert np.allclose(
+            self.base_values, other.base_values, atol=1e-6
         ), "Can't hstack explanations with different base values!"
```

Comment:
Would an Exception be better here?

```suggestion
        if not np.allclose(self.base_values, other.base_values, atol=1e-6):
            raise ValueError("Can't hstack explanations with different base values!")
```